hail/python/hailtop/hailctl/config/utils.py

```diff
@@ -0,0 +1,149 @@
+import json
+import tempfile
+from typing import Dict, List, Optional
+
+
+class InsufficientPermissions(Exception):
+    def __init__(self, message: str):
+        self.message = message
+
+
+async def already_logged_into_service() -> bool:
+    from hailtop.auth.auth import async_get_userinfo  # pylint: disable=import-outside-toplevel
+    try:
+        await async_get_userinfo()
+        return True
+    except Exception:
+        return False
+
+
+async def check_for_gcloud() -> bool:
+    from hailtop.utils import check_shell  # pylint: disable=import-outside-toplevel
+    try:
+        await check_shell("gcloud version")
+        return True
+    except Exception:
+        return False
+
+
+async def get_gcp_default_project(verbose: bool) -> Optional[str]:
+    from hailtop.utils import check_shell_output  # pylint: disable=import-outside-toplevel
+    try:
+        project, _ = await check_shell_output("gcloud config get-value project", echo=verbose)
+        project = project.strip().decode('utf-8')
+        return project
+    except Exception:
+        return None
+
+
+def get_default_region(supported_regions: List[str], cloud: str) -> Optional[str]:
+    if cloud == 'gcp':
+        if 'us-central1' in supported_regions:
+            return 'us-central1'
```

Comment:
Is this supposed to represent the default region in the deploy config? This isn't necessarily true for e.g. the australians. How about another endpoint like the supported regions?

If this doesn't represent the default region of the hail batch service, then I think we should not have a default and force the user to think about which region they really want.